chart: configure Helm v3 repositories

PaulFarver opened the pull request:
Currently when deploying the helm chart with v3 enabled and a repositories file is configured with configureRepositories, the repositories are only available to helm2, since helm3 has a different config path

This pull request adds logic to the deployment template to only configure repositories for the enabled versions of helm.

Addresses part of #186

stefanprodan commented:
The operator should copy the repositories file for each version on its own, I guess this is bug? @hiddeco can you confirm? relates to #164

PaulFarver commented:
My initial approach was actually to include the `--helm-repository-import` for every enabled version, but since the file is already mounted to the helm v2 config directory it seemed to be more in line with the design to just mount it to the helm v3 config directory as well.

hiddeco commented:
The changes in this PR are technically fine, given the new setup however, I think it would be wiser to mount the file outside of the expected (Helm) location and supply it to the `--helm-repository-import` flag, so that the operator takes the proper route and steps to import all repositories and update the indexes (the latter being the most important).

PaulFarver committed:
configure repositories for both helm v2 and v3

switch to configuring helm repositories with --helm-repository-import
when installing helm-operator helm chart

Signed-off-by: Paul Farver <paulfarver@gmail.com>

PaulFarver commented:
@hiddeco I have switched to a `--helm-repository-import` approach, but the logic taking care of the repository cache is still there in the same way as before.

hiddeco reviewed (approved):
Tested and confirmed working, thanks a lot @PaulFarver :stars:
